### ISSUE TITLE
Navigate through pages *and* layers

### ISSUE THREAD

**Issue by gyps** [xournalpp/xournalpp#1646]
I like the concept of using layers for overlay slides. To present such layered slides you can navigate through the slides using next/previous layer (ctrl-shift-left/right). As soon as one hits the last/first layer the desired behavior in this scenario would be to go to the next page instead. Currently next layer has no effect on the last layer. It should be easy to implement and from my perspective the expected behavior.

What do you think? Does anything speak against this?

**Comment by LittleHuba** [comment]
I'm not really happy with the solution for the border case where we switch pages, as this is not really intuitive to users not having read this ticket. Why don't you duplicate slides? That would have the same result but with all wished features already implemented.

**Comment by gyps** [comment]
> Why don't you duplicate slides? That would have the same result but with all wished features already implemented.

Not exactly, If you have lets say 10 overlays and on the 2nd overlay make a mistake then it is easy to edit the 2nd layer but if you need to fix the error in all 9 copies is at best very tediuous work. I did that for 15 heavily "overlayed" slides once and spend 90% of the time fixing errors in lower layers (I used a tool that did not support layers).

**Comment by LittleHuba** [comment]
Ahh so you see that as some kind of animated annotations using layers?

**Comment by gyps** [comment]
Yes, I like the mixture of LaTeX'd and hand written slides/overlays in my presentations (plus of course live annotations where helpful).

**Comment by LittleHuba** [comment]
Is it possible for you to share such a file? I have the feeling I do not really get your workflow yet.

Just a random thought, which I can't tell if it helps in this matter because of the potential missunderstanding: We will support referencing attachments in the future. Does this help you in any way with your issue?

**Comment by ea42gh** [comment]
Somewhat related to #1647

[BeamerikZ and Beamer](https://www.mimuw.edu.pl/~mskrzypczak/projects/beamerikz/ ) let me generated animated pdf files up to a point.

Importing such a pdf file as background works: I can navigate the slides and use xournalpp to annotate them. What layers might let me do is to enhance the animation by strategically overlaying
parts of a slide with a geometric shape colored the same as the background (e.g., uncovering
parts of an equation at a time during a presentation).

What would be nice to have is an integration of previous/next layer commands with beamers previous/next slider mechanisms which are tied to the arrow keys:
switch from layer to layer. Once there are no more layers on a page, go to the next page.

**Comment by mikeshulman** [comment]
I've been thinking about the same use case in the OP.  But wouldn't it work to just hit the 'next page' button when you get to the top layer?  In the worst case you forget that you're on the top layer already, nothing happens when you hit 'next layer', and you realize you have to hit 'next page' instead.

**Comment by rolandlo** [comment]
The suggested solution makes a lot of sense to me. We should implement it.

**Comment by camoz** [comment]
But please make it opt-in behavior. Yes, the use case sounds useful, but switching pages when clicking on "next layer" sounds rather unintuitive to me, even if there was only one layer... this sounds more as if it should be a special presentation mode or something.
